src/MobileDevice/widget/MobileDevice.ts

```diff
@@ -0,0 +1,126 @@
+import * as dojoDeclare from "dojo/_base/declare";
+import * as domConstruct from "dojo/dom-construct";
+import * as WidgetBase from "mxui/widget/_WidgetBase";
+import * as dojoLang from "dojo/_base/lang";
+
+type OnGetDeviceInformationOptions = "doNothing" | "showPage" | "callMicroflow";
+
+class MobileDevice extends WidgetBase {
+    private deviceIDAttribute: string;
+    private deviceTypeAttribute: string;
+    private appVersionVersionAttribute: string;
+    private appVersionBuildAttribute: string;
+    private onGetDeviceInformationMicroflow: string;
+    private onGetDeviceInformationOption: OnGetDeviceInformationOptions;
+    private onGetDeviceInformationPage?: string;
+    private mxObject: mendix.lib.MxObject;
+
+    update(mxObject: mendix.lib.MxObject, callback: () => void) {
+        this.mxObject = mxObject;
+        const warnings = this.validateProps();
+        if (warnings) {
+            domConstruct.create("div", {
+                class: "alert alert-danger",
+                innerHTML: warnings
+            }, this.domNode);
+        } else {
+            this.setUpWidgetDom();
+            this.connect(document, "deviceReady", dojoLang.hitch(this, this.onDeviceReady));
```

Comment:
Since i am checking now if it is a device, I have functions to `setUpEvents` and `removeEvents`